src/fondant/pipeline/lightweight_component.py

```diff
@@ -49,6 +53,49 @@ def image(cls) -> Image:
 
         return AppliedPythonComponent
 
+    def validate_return_types(base_component_cls, cls_implementation):
```

Comment:
Any reason to only check the return type instead of the full signature?

Reply:
Indeed no reason for it. Changed it.